### PR TITLE
ContrastCurveModule

### PR DESCRIPTION
**Pull request by 404NotFoundv2**
Here is the implementation of multiprocessing for the ContrastCurveModule. I tested it both on Bluesky & Euler.

The input arrays containing the -_images_- and -_psf_- are saved in numpy binary format in the -_working_place_-. These numpy files are loaded in each subprocess. When all the processes have finished, those files are removed.

The number of processes is set by the -_cpu_- in the config file. As each process gets an copy of this in memory, the cpu should not be set too high.

Also the '.' has been replaced by a percentage sign, but this can easily be, if required, removed.

Lastly, the result in each process is saved in a queue. This object is guarantees safe handling of information between multiple threads/processes.

I left some comments in there to understand what each part does.

### CODE REVIEW

**Comment by tomasstolker** [comment]
Thanks for the efforts, @404NotFoundv2! It looks good to me.

The 'dot' was used as progress indicator because processes may not finish in order. Is the percentage indicator always increasing?

**Comment by 404NotFoundv2** [comment]
Great!
The dot was written to the output when a magnitude at a single position had been found/converged, correct?

This is how what I've done now:
Say _cpu=5_, number of positions to calculate=40
It starts the calculating the first 5 positions. When all 5 are finished, the percentage sign jumps from 0% to 12.5%. Thus, if cpu would be set to 40, it jumps from 0 to 100%.

The dot would output a '.' when a single process is finished, but usually the 5 processes finish at approximately the same time.

TL;DR, Yes it increases. When #*_cpu_* process is finished, it increases

**Comment by tomasstolker** [comment]
OK good, thanks!
